src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,223 @@
+            identifier = identifier.replace("/document/", "/resource/content?uri=");
+            identifier = identifier.replace("https://isidore.science/", "https://api.isidore.science/");
+            return new URL(identifier);
+        } else {
+            throw new FetcherException("Could not construct url for ISIDORE");
+        }
+    }
+
+    @Override
+    public Parser getParser() {
+        return this.parser;
+    }
+
+    private BibEntry xmlItemToBibEntry(Document document) {
+        Element itemElement = document.getDocumentElement();
+        return new BibEntry(getType(itemElement.getElementsByTagName("types").item(0).getChildNodes()))
+                .withField(StandardField.TITLE, itemElement.getElementsByTagName("title").item(0).getTextContent())
+                .withField(StandardField.AUTHOR, getAuthor(itemElement.getElementsByTagName("enrichedCreators").item(0)))
+                .withField(StandardField.YEAR, itemElement.getElementsByTagName("date").item(0).getChildNodes().item(1).getTextContent().substring(0, 4))
+                .withField(StandardField.JOURNAL, getJournal(itemElement.getElementsByTagName("dc:source")))
+                .withField(StandardField.PUBLISHER, getPublishers(itemElement.getElementsByTagName("publishers").item(0)))
+                .withField(StandardField.DOI, getDOI(itemElement.getElementsByTagName("ore").item(0).getChildNodes()))
+                .withField(StandardField.URL, this.URL);
+    }
+
+    private String getDOI(NodeList list) {
+        for (int i = 0; i < list.getLength(); i++) {
+            if (list.item(i).getTextContent().contains("DOI:")) {
+                return list.item(i).getTextContent().replace("DOI: ", "");
+            }
+            if (list.item(i).getTextContent().contains("doi:")) {
+                return list.item(i).getTextContent().replace("info:doi:", "");
+            }
+        }
+        return "";
+    }
+
+    // Get the type of the document, ISIDORE only seems to have select types, also their types are different to
+    // those used by JabRef.
+    private EntryType getType(NodeList list) {
+        for (int i = 0; i < list.getLength(); i++) {
+            String type = list.item(i).getTextContent();
+            if (type.contains("article") || type.contains("Article")) {
+                return StandardEntryType.Article;
+            }
+            if (type.contains("thesis") || type.contains("Thesis")) {
+                return StandardEntryType.Thesis;
+            }
+            if (type.contains("book") || type.contains("Book")) {
+                return StandardEntryType.Book;
+            }
+        }
+        return StandardEntryType.Misc;
+    }
+
+    // Gets all the authors, separated with the word "and"
+    // For some reason the author field sometimes has extra numbers and letters.
+    private String getAuthor(Node itemElement) {
+        boolean singleAuthor = true;
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Please use `new StringJoiner(" and ")`. Then you can avoid the if (singleAuthor) check in lines 145 to 149.